app/org/sagebionetworks/bridge/dynamodb/DynamoExternalIdentifier.java

```diff
@@ -0,0 +1,85 @@
+package org.sagebionetworks.bridge.dynamodb;
+
+import org.sagebionetworks.bridge.models.accounts.ExternalIdentifier;
+import org.sagebionetworks.bridge.models.studies.StudyIdentifier;
+
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBAttribute;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBHashKey;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBRangeKey;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTable;
+
+/**
+ * Implementation of external identifier.
+ */
+@DynamoDBTable(tableName = "ExternalIdentifier")
+public class DynamoExternalIdentifier implements ExternalIdentifier {
+
+    private String identifier;
+    /**
+     * DynamoDB will not filter a query on the value of the hash key, so we copy this 
+     * value to another column where we can filter on it. This is not exposed outside
+     * of the DAO.
+     */
+    private String filterableIdentifier;
```

Comment:
I'll switch to use of range key. If that's not good enough, we can probably change later.